### ISSUE TITLE
Comparing versions without metadata

### ISSUE THREAD

**Issue by mkpaz** [skuzzle/semantic-version#2]
Sadly, there're no way to increment pre-release to stable version, e.g. `1.0.0-SNAPSHOT >>> 1.0.0`.
I have a situation, there're minimal and required version, and when I compare them I don't wan't to take pre-release/metadata into account.

```
Version requiredVersion = Version.parseVersion("1.0.0");
Version currentVersion = Version.parseVersion("1.0.0-SNAPSHOT");

// currentVersion won't ever match requiredVersion 
System.out.println(currentVersion.isGreaterThan(requiredVersion) | currentVersion.equals(requiredVersion));
```

The only wait to do so at the moment is to compare major, minor, and patch versions separately, which isn't very convenient. There're bunch of `with***()` and `next***()` methods, would be very nice to have something like `toStable()` or `withoutPrerelease()`.

P.S. Also something like `isGreaterOrEqual()` and `isLesserOrEqual()` would be a great help.

**Comment by skuzzle** [comment]
Thanks for your feedback. 
For now, you can go with `version.withPreRelease("")` which simply removes the pre-release identifier.  To be semantically equivalent to the other `next***` methods, you might want to also drop the build meta data by chaining the methods like `version.withPreRelease("").withBuildMetadata("")` 
I'll consider to add a dedicated method to make this more readable and to better convey the intention to the reader.

In your case, you could also use `compare` like in the following snippet. I agree though that it would be nice to have dedicated methods for inclusive comparisons.

```java
Version requiredVersion = Version.parseVersion("1.0.0");
Version currentVersion = Version.parseVersion("1.0.0-SNAPSHOT");

System.out.println(currentVersion.compareTo(requiredVersion) >= 0);
```

**Comment by mkpaz** [comment]
Wow, thanks a lot!
A bit not obvious but works well enough.

**Comment by skuzzle** [comment]
I'll reopen this to track the progress of implementing furhter convenience methods.

I'm not yet sure on how to handle the build meta data identifier when calling `toStable`. To comply with the `next***` methods, build meta data should be dropped but strictly speaking, it doesn't have influence on whether the version is stable or not. So it is considerable to not drop it.

**Comment by skuzzle** [comment]
I've added the following new convenience methods:
* `toStable()`: Removes pre-release and build meta data
* `isLowerThanOrEqualTo(other)`: as the name says
* `isGreaterThanOrEqualTo(other)`: as the name says

Changes will be released soon as `2.1.0`. There will be **no** backport of these new features to the java 6,7,8 version of this library

**Comment by skuzzle** [comment]
Released as 2.1.0
